File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/command/argument/CommandArgFloat.java
package pize.tests.voxelgame.main.command.argument;

import pize.tests.voxelgame.server.Server;
import pize.tests.voxelgame.main.command.source.CommandSource;

public class CommandArgFloat extends CommandArg{
    
    // Результат парсинга
    private float number;
    
    @Override
    public int parse(String remainingChars, CommandSource source, Server server){
        // Разделяем оставшуюся часть команды на части
        final String[] args = remainingChars.split(" ");
        
        // Если количество частей меньше 1 (число), завершить парсинг
        if(args.length < 1)
            return 0;
        
        // устанавливаем число
        number = Float.parseFloat(args[0]);
        return args[0].length();
    }
    
    public float getFloat(){
        return number;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/command/argument/CommandArgFloat.kt
package pize.tests.voxelgame.main.command.argument

import pize.tests.voxelgame.main.command.source.CommandSource
import pize.tests.voxelgame.server.Server

class CommandArgFloat : CommandArg() {
    // Результат парсинга
    var float = 0f
        private set

    override fun parse(remainingChars: String, source: CommandSource, server: Server): Int {
        // Разделяем оставшуюся часть команды на части
        val args = remainingChars.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

        // Если количество частей меньше 1 (число), завершить парсинг
        if (args.size < 1) return 0

        // устанавливаем число
        float = args[0].toFloat()
        return args[0].length
    }
}
